chains/evm/calls/events/listener.go

```diff
@@ -38,8 +36,42 @@ func NewListener(client ChainClient) *Listener {
 	}
 }
 
-func (l *Listener) FetchDepositEvent(event RetryEvent, bridgeAddress common.Address, blockConfirmations *big.Int) ([]events.Deposit, error) {
-	depositEvents := make([]events.Deposit, 0)
+func (l *Listener) UnpackDeposit(abi abi.ABI, data []byte) (*Deposit, error) {
```

Comment:
Minor nitpick. Can you please make it private and below the function where it is used